src/consensus/pbft_manager.cpp

```diff
@@ -582,7 +668,63 @@ bool PbftManager::stateOperations_() {
   return resetRound_();
 }
 
-void PbftManager::updateNextVotedValue_() {
+bool PbftManager::updateSoftVotedBlockForThisRound_() {
+  if (!soft_voted_block_for_this_round_.second) {
+    auto round = getPbftRound();
+
+    auto soft_votes = getVotesOfTypeFromVotesForRoundAndStep_(soft_vote_type, votes_, round, 2,
+                                                              std::make_pair(NULL_BLOCK_HASH, false));
+    auto voted_block_hash_with_soft_votes = blockWithEnoughVotes_(soft_votes);
+
+    auto batch = db_->createWriteBatch();
+    db_->addPbftMgrVotedValueToBatch(PbftMgrVotedValue::soft_voted_block_hash_in_round,
+                                     voted_block_hash_with_soft_votes.voted_block_hash, batch);
+    db_->addPbftMgrStatusToBatch(PbftMgrStatus::soft_voted_block_in_round, voted_block_hash_with_soft_votes.enough,
+                                 batch);
+    if (voted_block_hash_with_soft_votes.enough &&
+        voted_block_hash_with_soft_votes.voted_block_hash != NULL_BLOCK_HASH) {
+      db_->addSoftVotesToBatch(round, voted_block_hash_with_soft_votes.votes, batch);
+    }
+    db_->commitWriteBatch(batch);
+
+    soft_voted_block_for_this_round_ =
+        std::make_pair(voted_block_hash_with_soft_votes.voted_block_hash, voted_block_hash_with_soft_votes.enough);
+
+    if (soft_voted_block_for_this_round_.first != NULL_BLOCK_HASH && soft_voted_block_for_this_round_.second) {
+      // Have enough soft votes for a value other other than NULL BLOCK HASH...
+
+      if (state_ == finish_polling_state) {
+        LOG(log_dg_) << "Node has seen enough soft votes voted at " << soft_voted_block_for_this_round_.first
+                     << ", regossip soft votes. In round " << round;
+        if (auto net = network_.lock()) {
+          net->onNewPbftVotes(move(voted_block_hash_with_soft_votes.votes));
+        }
+      }
+
+      return true;
+    }
+  }
+  return false;
+}
+
+void PbftManager::initializeVotedValueTimeouts_() {
+  time_began_waiting_next_voted_block_ = std::chrono::system_clock::now();
+  time_began_waiting_soft_voted_block_ = std::chrono::system_clock::now();
+  if (soft_voted_block_for_this_round_.first != NULL_BLOCK_HASH && soft_voted_block_for_this_round_.second) {
+    last_soft_voted_value_ = soft_voted_block_for_this_round_.first;
+  }
+}
+
+void PbftManager::setLastSoftVotedValue(blk_hash_t soft_voted_value) { checkSoftVotedValueChange_(soft_voted_value); }
```

Comment:
The function did the same thing with checkSoftVotedValueChange_, could remove and call checkSoftVotedValueChange_ directly

Reply:
My thinking was to add a guard once we have one (or do we already have one) so that can ensure this public function is only called from within a test.